src/dmd/dtemplate.d

```diff
@@ -6597,7 +6597,7 @@ extern (C++) class TemplateInstance : ScopeDsymbol
                     Dsymbol s2 = s.getType().toDsymbol(sc);
                     if (!s2)
                     {
-                        error("`%s` is not a template declaration, it is a %s", id.toChars(), s.kind());
+                        error("`%s` is not a template declaration, it is a type", id.toChars());
```

Comment:
Technically, it is an alias, as type parameters are accepted by alias. However, it could also be an alias to some variable, making the error message still wrong.
I suggest testing to see of s2 is an alias. If it is, then resolve the alias to what it really is using `toAlias()`, then use the original error message.

Reply:
I haven't found any examples where that particular error is triggered for something different than a type. It looks like there is another branch for non-types [1] . In addition, using the original error message will never specify that a type is used instead of a template declaration because `kind` never returns "type" for type AST nodes (instead it returns the actual name of the type for basic types and `struct`, `class` etc. for user defined types)

[1] https://github.com/dlang/dmd/blob/master/src/dmd/dtemplate.d#L6625